src/protocol/payload/addr.rs

```diff
@@ -86,6 +86,10 @@ impl NetworkAddr {
     pub(super) fn decode_without_timestamp<B: Buf>(bytes: &mut B) -> io::Result<Self> {
         let services = u64::from_le_bytes(read_n_bytes(bytes)?);
 
+        if bytes.remaining() < 16 {
```

Comment:
We could consider using `read_n_bytes` everywhere instead, since it checks for the length. Not sure if it would be slower than the built in `get_*` methods.

Reply:
yeah, I had the same thoughts; we might want to align this at one point in time, but I'm ok with both options